### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by y-satskyi**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,13 +1,18 @@
-def format_linter_error(error: dict) -> dict:
-    # write your code here
-    pass
+def format_linter_error(error):
+    return {"line": error["line_number"],
+            "column": error["column_number"],
+            "message": error["text"] ,
+            "name": error["code"],
+            "source": "flake8"}
```
Comment (IvanRamyk) [inline app/main.py:6]:
```suggestion
    return {
        "line": error["line_number"],
        "column": error["column_number"],
        "message": error["text"] ,
        "name": error["code"],
        "source": "flake8"
    }
```

File: app/main.py [hunk 2]
```diff
@@ -1,13 +1,18 @@
...
 
 
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
-    pass
+    return {"errors": [format_linter_error(error) for error in errors],
+            "path": file_path,
+            "status": ("passed" if len(errors) == 0 else "failed")}
```
Comment (IvanRamyk) [inline app/main.py:12]:
```suggestion
    return {
        "errors": [format_linter_error(error) for error in errors],
        "path": file_path,
        "status": ("passed" if len(errors) == 0 else "failed")
    }
```

File: app/main.py [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
 
 
-def format_linter_report(linter_report: dict) -> list:
-    # write your code here
-    pass
+def format_linter_report(linter_report):
+    return [
+        format_single_linter_file(file_path=key, errors=value)
+        for key, value in linter_report.items()]
```
Comment (IvanRamyk) [inline app/main.py:18]: fix here as well

**Review by IvanRamyk — CHANGES_REQUESTED** [review]
A few code-style suggestions

**Comment by y-satskyi** [comment]
I fixed

**Comment by y-satskyi** [comment]
Can you check again?
I have already done yesterday

**Pull request closed without merging**